app/src/main/java/zapsolutions/zap/fragments/SendBSDFragment.java

```diff
@@ -555,6 +556,25 @@ public void onClick(View v) {
         return view;
     }
 
+    public void createLightningDialog(BaseAppCompatActivity activity, PayReq paymentRequest, String invoice) {
+        Intent intent = new Intent();
```

Comment:
I would make those static and move `SendBSDFragment sendBottomSheetDialog = new SendBSDFragment();` inside here, and then return the `SendBSDFragment` instance to the caller.
And move the `.show()` outside, because this should only create the instance, and then the caller can decide when to show.

Reply:
Done.
I also readded the finishAffinity stuff because when testing I found out again why I needed it in the first place.
Without this if you are already on the home screen, then minimize the app and then start it with an URI scheme the following happens:

It still has the minimized HomeActivity in the backstack. It then opens the Landing activity.
It then opens a new HomeActivity. Now it starts to establish the connection to LND on the new HomeActivity. And a few milliseconds later it starts the on destroy method of the old HomeActivity. This again closes the connection to LND and we are unconnected, but without showing an error...

FinishAffinity() makes sure all the onDestroy methods of existing Activites are executed before we start the new Activity.

Reply:
Cool, thanks for explanation!